### PR TITLE
feat:CO-21 : container query instead of media query

### PR DESCRIPTION
**Pull request by arjun-at-roam**
## PR Area

- [x] Frontend

## Type of change

_What kind of change does this PR introduce?_

- [x] Feature

## Description

Problem: The widget is intended to be embedded into a div container of unknown width on the merchant website. The original widget responsive design was based on the overall width of the view port. This design caused the widget to not be rendered correctly when it is embedded in a small merchant container.
Solution: Instead of using view port width as a reference to trigger responsive design, use parent container width.

## Security & risks

- [ ] Adds new third party libraries
- [ ] Adds new `.env` variables
- [ ] Integrates with Third-party apis (dependency must be resilient)
- [ ] Has specific [OWASP security concerns](https://owasp.org/www-project-top-ten/) - list them: (also see [Fixing OWASP Top 10](https://nodegoat.herokuapp.com/tutorial))

## Pre-publish checklist

- [x] I have assigned myself to the ticket
- [x] I have performed a self-review of my code
- [x] I have performed relevant browser testing
- [ ] I have performed keyboard testing of new or modified interactive components

#### Evidence screenshots

Before:
image

After:
image

## Reviewer checklist

- [ ] I have run the code (if applicable)
- [ ] I have reviewed and run any tests (if applicable)

### CODE REVIEW

File: public/demo.html [hunk 1]
```diff
@@ -10,34 +10,39 @@
     <!-- This demo file is used to test both the production bundles after running yarn build -->
     <h1>Humm widgets cards online</h1>
 
-    <h2>AU default widget</h2>
-    <script src="./humm-widgets-cards-au.umd.js?productPrice=400&merchantId='aaa143af-a96b-4887-bcd2-85960da544ea'"></script>
+    <div
+      class="resizeWidget"
+      style="resize: horizontal; overflow: auto; width: fit-content"
+    >
```
Comment (arjun-at-roam) [inline public/demo.html:16]: horizontally resizable parent div to test responsiveness
image

File: src/components/buttons/Button.vue [hunk 2]
```diff
@@ -90,17 +90,12 @@ export default defineComponent({
...
-    max-width: 138px;
-
-    @media (min-width: 430px) {
-      letter-spacing: -0.01em;
-      font-size: 12px;
-      font-weight: 500;
-      width: auto;
-      min-width: 90px;
-      min-height: 20px;
-    }
```
Comment (arjun-at-roam) [inline src/components/buttons/Button.vue:103]: Unnecessary media query as the client designer doesnt want the button to change size based on width
Before:
image
After:
image

File: src/components/dataDisplay/Card.vue [hunk 3]
```diff
@@ -28,10 +28,6 @@ export default defineComponent({
 
   &--auto {
     width: 34px;
-
-    @media (min-width: 430px) {
-      width: 35px;
-    }
```
Comment (arjun-at-roam) [inline src/components/dataDisplay/Card.vue:34]: this media query is overkill for 1 px

File: src/providers/Responsive.vue [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      this.$emit('resize', { width })
+    },
+    initObserver() {
+      const observer = new ResizeObserver(this.onResize)
+      observer.observe(this.$refs.box as Element)
+      this.observer = observer
+    },
+  },
+})
+</script>
```
Comment (arjun-at-roam) [inline src/providers/Responsive.vue:37]: The most important bits are here ☝️ 
using ResizeObserver get parent width (`clientWidth`)

if the `clientWidth` > 410 then `isSizeLarge` is `true`

File: src/styles/cards.scss [hunk 5]
```diff
@@ -21,12 +21,8 @@
   }
 
   &--logo {
-    width: 75px;
+    min-width: 75px;
     margin-top: 3px;
-
-    @media (min-width: 430px) {
-      width: 107px;
-    }
```
Comment (arjun-at-roam) [inline src/styles/cards.scss:29]: This is a collateral change. Give the logo a min-width.
image

File: src/styles/widget.scss [hunk 6]
```diff
@@ -7,7 +7,7 @@
   position: relative;
   filter: drop-shadow(0 1px 5px rgba(0, 0, 0, 0.1));
 
-  @media (min-width: 430px) {
+  &[data-size-large='true'] {
```
Comment (arjun-at-roam) [inline src/styles/widget.scss:10]: This is how we'll be using `data-size-large` when it is set to true/false in `WidgetContent.vue`
image

File: src/components/dataDisplay/Card.vue [hunk 7]
```diff
@@ -28,10 +28,6 @@ export default defineComponent({
...
   }
```
Comment (arjun-at-roam) [inline src/components/dataDisplay/Card.vue:31]: this media query is overkill for 1 px

**Review by jpyepez-roam — APPROVED** [review]
LGTM! Nice job cleaning up extra media queries and keeping the codebase clean!

File: src/providers/Responsive.vue [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    onResize() {
+      const box = this.$refs.box as Element
+      const width = box.clientWidth
+      this.width = width
+
+      if (width > 410) {
+        this.isSizeLarge = true
+      } else {
+        this.isSizeLarge = false
+      }
```
Comment (antoninroam) [inline src/providers/Responsive.vue]: **nit:** just a tip, you can shorten it like that: `this.isSizeLarge = width > 410`

**Review by antoninroam — APPROVED** [review]

File: src/providers/Responsive.vue (hunk 8 above)
Comment (arjun-at-roam) [reply src/providers/Responsive.vue]: 🤤 that's def what I want. Changing it to your suggestion.

**Pull request merged**